README.rst

```diff
@@ -246,3 +246,113 @@ Kør nedenstående kommando for at autogenerere dokumentationen::
 
 Dokumentation kan nu findes ved at åbne filen
 ``/sti/til/mora/docs/out/index.html``.
+
+Licens og Copyright
+-------------------
+
+Copyright OS2 Offentlig Digitaliseringsfællesskab. 
```

Comment:
```suggestion
```

Vi bør ikke påstå at ophavsretten tilhører OS2, med mindre det faktisk er tilfældet, og kildekoden selv påstår at ophavsretten tilhører Magenta.

Reply:
Krav fra OS2 via Alex via #24655 med henvisning til:
https://faq.os2.eu/sites/faq.os2.eu/files/uploaded_files/licensvalg_open_source_2.pdf
"""Derudover bør filen readme.txt i projektet have en header, der indeholder: ”Copyright OS2 Offentlig Digitaliseringsfællesskab. Dette værk er licenseret under open
source licensen Mozilla Public License 2.0. Læs mere i license.txt filen.” """

Reply:
Nu er det ikke mit bord som sådan — men jeg synes ikke vi bør skrive noget der er decideret forkert på dette punkt, hvad enten de beder om det eller ej. Koden bliver jo ikke deres blot fordi vi skriver at den er det — og det er end ikke sikkert at den rent faktisk er vores selv om vi påstår det mange andre steder. Det kræver jo en gennemgang af de kontrakter vi skrev den under…

Reply:
Hvis ikke der eksplicit står noget i kontrakten om, at ophavsretten til koden overdrages til modtageren (eller hvem det nu er), tilhører ophavsretten til koden Magenta.

Reply:
Hvad med:

```suggestion
Dette er et OS2 projekt. Ophavsretten tilhører de individuelle bidragydere.
```

Det er i øvrigt langt fra alle open source projekter der kræver at ophavsretten overdrages — og de der gør, sørger netop for papirarbejde for at udføre det i praksis.

Reply:
Hvis OS2 ønsker ophavsretten til OS2-kode overdraget for at betragte det som et OS2-projekt kan vi godt det, men det vil kræve en officiel erklæring eller kontrakt fra vores side, for uden en sådan erklæring, tilhører ophavsretten til al software skrevet af Magenta-ansatte (ikke overraskende) Magenta.
I denne kontrakt kunne vi evt. stipulere, at OS2 lover aldrig at ændre licensen til en ikke-OSI-godkendt licens.
Uanset at OS2 har ophavsretten, vil vi altid have lov til at bruge source koden under MPL, fordi koden så at sige overdrages til os under MPL i det øjeblik, vi skriver den.

Reply:
Så konklusionen er, at koden ligger hos dem, men ophavsretten ændres ikke, mm. det bliver ekspliciteret i en kontrakt. Hvis I er enige, så behøver vi ikke gøre mere.

Reply:
:+1:

Reply:
Rasmus Frey siger, at udover signalværdien er årsagen til at koden skal ligge hos OS2 netop det faktum at OS2 gerne vil beskytte sig mod skumle folk og firmaer, der kunne finde på at skifte licensen. Hvordan det forholder sig, hvis nogen skulle få lyst til det - skal der fx opnås konsensus hos alle bidragydere? - ved vi ikke. Hvis vi skal have vendt hver en sten, skal vi kontakte en jurist. Indtil videre er jeg tryg ved, at dette PR bliver godkendt.

Reply:
Bortset fra, at hvis ophavsretten skal overdrages til OS2, bør denne header i samtlige kodefiler også ændres:
```
#
# Copyright (c) 2017-2018, Magenta ApS
#
```

Reply:
1) Licensen i sig selv forhindrer selv forhindrer fratagelse af rettigheder. Selv hvis vi, fx, skulle ændre licensen vil hidtil frigivne ændringer forblive frie. Vi kan allerhøjst frigive den til andre under anderledes vilkår, eller give ny kode til andre under andre vilkår.
2) Ja, en licensændring kræver tilsagn fra _alle_ der har ejerskab til koden. I dette tilfælde er det udelukkende os.
3) Efter min ydmyge overbevisning burde han stille disse spørgsmål til en advokat eller jurist — vi har holdninger og idéer, men til syvende og sidst er det ren jura.

Reply:
Det er værd at bemærke, at lige så lidt som vi, hvis vi beholder ophavsretten, vil kunne forhindre OS2 i at arbejde videre med kode, de allerede har fået - lige så lidt vil OS2 kunne forhindre os i at arbejde videre med kode, vi har skrevet.

Hvis vi bliver uenige med dem og de ønsker at få en anden leverandør til at stå for den fremtidige udvikling under en proprietær licens, vil vi stadig have al koden under MPL og have fuld ret til at udvikle mere ved den. Vi irisikerer altså ikke at miste noget ved en sådan overdragelse.

Til gengæld er det en god idé at spørge en advokat om, hvad der skal til af bindende erklæringer, for at ophavsretten _faktisk_ er overdraget. Jeg er sikker på, at det ikke er nok at vi opdaterer README-filen.

Reply:
> Ad 3 - det vil han også gøre.

Hvad gør vi indtil da?

Reply:
Vi er i færd med at smede en formulering, der stipulerer, at OS2's ophavsret bortfalder, hvis de på noget tidspunkt skulle finde på at ændre til alt andet end en Copyleft licens. Det bliver Magentas politik fremover ved lignende sager. Indvendinger?

Reply:
Så dette PR afventer den formulering

Reply:
Det lyder fint!

Reply:
> Vi er i færd med at smede en formulering, der stipulerer, at OS2's ophavsret bortfalder, hvis de på noget tidspunkt skulle finde på at ændre til alt andet end en Copyleft licens. Det bliver Magentas politik fremover ved lignende sager. Indvendinger?

Det lyder i mine øren ret mærkeligt — som jeg har forstået det er ophavsret en form for ejerskab, og man kan da ikke umiddelbart betinge en overdragelse på fremtidige handlinger, kan man? Jeg ville snakke med en advokat om den slags…

I øvrigt har vi stort set allerede givet dem de rettigheder, qua MPLen.

Reply:
Det hele er i proces og der vil komme jura-øjne på, så med lidt tålmod skal det hele nok falde fint på plads

Reply:
Svar fra et OS2-bestyrelsesmedlem: 

> jeg forstår jeres bekymringer og det er dejligt dine medarbejdere er opmærksomme på denne problemstilling. Vi vil ikke have gode FOSS produkter lukket inde i closed source licenser.
> 
> Det er korrekt at regttighedshaveren til enhver tid kan ændre licensen så det kan OS2 lige såvel som Magenta – hvis vedkommende part er indehaver af IP rettighederne.
> 
> Denne ændring af licenser kan dog alene foretages for Nye/fremtidige versioner. Den nuværende open source licens som produktet er udgivet under kan ikke ændres for den udgivne version.
> 
> Hvis OS2MO koden overdrages til OS2 fællesskaber og OS2 fællesskabet (højere magter forbyde det)  vil ændre licenser til en closed source licens, kan dette alene ske for en ny udgave af produktet/ny version af koden. Men så er det jo heller ikke Magentas kode det sker for, men en senere version/revision.
> 
> Den version af koden som Magenta overdragede vil for altid have den licens som Magenta gav den, da Magenta havde rettighederne.
> 
> Der er således ingen risiko for Magenta koden i forhold til at den bliver lavet til closed source.